src/operator/contrib/count_sketch.cu

```diff
@@ -129,8 +129,8 @@ inline void CountSketchForward(const Tensor<gpu, 2, DType> &out,
                                     nthreads, out_ptr+bstart*out_dim, h_ptr,
                                     s_ptr, in_ptr+bstart*in_dim, batchlen,
                                     in_dim, out_dim);
-    MSHADOW_CUDA_POST_KERNEL_CHECK(sketch_forward_kernel);
-    // cudaThreadSynchronize();
+    cudaError_t err = cudaDeviceSynchronize();
+    CHECK_EQ(err, cudaSuccess) << "Error occured! CUDA: " << cudaGetErrorString(err);
```

Comment:
why do we need to replace this? is it to improve the error message ?

Reply:
It's hard to explain within a short comment, plus all contexts are in the code.

Reply:
@anirudh2290 the original tracking issue is #10988, and I can confirm that I've seen multiple seeds that can reproduce the issue on my dev machine. With this fix it's passing more than 15000 times of tests even on the ones that failed previously.

Reply:
you mentioned that not replacing with `cudaDeviceSynchronize` caused certain values in result ndarray to be zeros causing test to fail. The stream synchronize should happen as part of `Wait` irrespective. This not happening seems to be a more serious problem and good to open an issue on this.

Reply:
You're right about that, we can discuss in person tomorrow about this so that we create a suitable issue for this.